xamarin/Xamarin.Forms
# [Core] Propagate BindingContext to GradientBrush

### Description of Change ###

Propagate BindingContext to GradientBrush.

### Issues Resolved ### 

- fixes #12339 

### API Changes ###

 None

### Platforms Affected ### 

- Core/XAML (all platforms)

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

<img width="525" alt="Captura de pantalla 2020-10-27 a las 11 22 20" src="https://user-images.githubusercontent.com/6755973/97289481-8a7b1980-1847-11eb-9fe0-96318c4f3bf1.png">

### Testing Procedure ###
Launch Core Gallery and navigate to the issue 12339. If you can see the gradient below, the test has passed.

### PR Checklist ###
